Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/SpecialZone.hs

```diff
@@ -0,0 +1,218 @@
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Dashboard.ProviderPlatform.SpecialZone
+  ( module Dashboard.ProviderPlatform.SpecialZone,
+    module Reexport,
+  )
+where
+
+import Dashboard.Common as Reexport
+import Data.Aeson
+import Kernel.External.Maps (LatLong)
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto hiding (Point)
+import Kernel.Types.APISuccess
+import Kernel.Types.App
+import Kernel.Types.Id
+import Kernel.Utils.JSON
+import Servant hiding (Summary)
+
+data SpecialZoneEndpoint
+  = CreateSpecialZoneEndpoint
+  | UpdateSpecialZoneEndpoint
+  | DeleteSpecialZoneEndpoint
+  deriving (Show, Read)
+
+derivePersistField "SpecialZoneEndpoint"
+
+---------------------------------------------------------
+-- crud special zone --------------------------------------
+
+type RegionLookupAPI =
+  "lookup"
+    :> "regions"
+    :> Header "token" Text
+    :> MandatoryQueryParam "minLatLng" LatLong
+    :> MandatoryQueryParam "maxLatLng" LatLong
+    :> Get '[JSON] [SpecialZone]
+
+type CreateSpecialZoneAPI =
+  "create"
+    :> Header "token" Text
+    :> ReqBody '[JSON] SpecialZoneAPIEntity
+    :> Post '[JSON] APISuccess
+
+type UpdateSpecialZoneAPI =
+  "update"
+    :> Header "token" Text
+    :> ReqBody '[JSON] SpecialZone
+    :> Post '[JSON] APISuccess
+
+type DeleteSpecialZoneAPI =
+  "delete"
+    :> Header "token" Text
+    :> MandatoryQueryParam "id" (Id SpecialZone)
+    :> Delete '[JSON] APISuccess
+
+data SpecialZone = SpecialZone
+  { id :: Id SpecialZone,
+    name :: Text,
+    categoryCode :: Category,
+    geoJson :: ShapeFile,
+    city :: Text,
+    state :: Text,
+    createdAt :: UTCTime,
+    updatedAt :: UTCTime
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
+data SpecialZoneAPIEntity = SpecialZoneAPIEntity
+  { name :: Text,
+    categoryCode :: Category,
+    geoJson :: ShapeFile,
+    city :: Text,
+    state :: Text
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
+data EntryExit = EntryExit
+  { id :: Id EntryExit,
+    specialZoneId :: Id SpecialZone,
+    _type :: EntryExitType,
+    lat :: Double,
+    lon :: Double,
+    area :: Maybe Text,
+    address :: Maybe Text,
+    createdAt :: UTCTime,
+    updatedAt :: UTCTime
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
+data EntryExitType = Entry | Exit deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema, Read)
+
+data Category
```

Comment:
Can you add comment on what classifies as `Sure` and `Unsure` for future reference

Reply:
Seems like this type should be wrapper in application using special zone service and not in special zone service itself